pkg/autoupdate/autoupdate.go

```diff
@@ -0,0 +1,229 @@
+package autoupdate
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/blang/semver"
+
+	"github.com/golang/glog"
+	"github.com/openshift/cluster-version-operator/lib/resourceapply"
+	"github.com/openshift/cluster-version-operator/pkg/apis/clusterversion.openshift.io/v1"
+	clientset "github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned"
+	"github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned/scheme"
+	informersv1 "github.com/openshift/cluster-version-operator/pkg/generated/informers/externalversions/clusterversion.openshift.io/v1"
+	listersv1 "github.com/openshift/cluster-version-operator/pkg/generated/listers/clusterversion.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	coreclientsetv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	// maxRetries is the number of times a machineconfig pool will be retried before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the times
+	// a machineconfig pool is going to be requeued:
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
+	maxRetries = 15
+
+	workQueueKey = "%s/%s"
```

Comment:
nit: doesn't feel this const is very useful and looks like it's only used in one place...

Reply:
:+1: